core/src/main/java/com/amplifyframework/core/store/EncryptedKeyValueRepository.kt

```diff
@@ -17,93 +17,193 @@ package com.amplifyframework.core.store
 
 import android.content.Context
 import android.content.SharedPreferences
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
 import androidx.annotation.VisibleForTesting
 import androidx.security.crypto.EncryptedSharedPreferences
 import androidx.security.crypto.EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV
 import androidx.security.crypto.EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
 import androidx.security.crypto.MasterKeys
+import com.amplifyframework.core.Amplify
 import java.io.File
+import java.security.KeyStore
 import java.util.UUID
 
-class EncryptedKeyValueRepository(
+class EncryptedKeyValueRepository @VisibleForTesting constructor(
     private val context: Context,
-    private val sharedPreferencesName: String
+    private val sharedPreferencesName: String,
+    private val defaultMasterKeySpec: KeyGenParameterSpec,
+    private val amplifyMasterKeySpec: KeyGenParameterSpec,
+    private val fileFactory: (dir: File, fileName: String) -> File
 ) : KeyValueRepository {
 
-    @VisibleForTesting
-    internal val sharedPreferences: SharedPreferences by lazy {
+    constructor(context: Context, sharedPreferencesName: String) : this(
+        context = context,
+        sharedPreferencesName = sharedPreferencesName,
+        defaultMasterKeySpec = getDefaultMasterKeySpec(),
+        amplifyMasterKeySpec = getAmplifyMasterKeySpec(),
+        fileFactory = { dir, fileName -> File(dir, fileName) }
+    )
+
+    private val sharedPreferences by lazy { getOrCreateSharedPreferences() }
+
+    override fun put(dataKey: String, value: String?) = edit { putString(dataKey, value) }
+    override fun get(dataKey: String): String? = sharedPreferences.getString(dataKey, null)
+    override fun remove(dataKey: String) = edit { remove(dataKey) }
+    override fun removeAll() = edit { clear() }
+
+    private inline fun edit(crossinline block: SharedPreferences.Editor.() -> Unit) = with(sharedPreferences.edit()) {
```

Comment:
Do we have a use case to apply set of changes atomically? While each edit operation is atomic, using multiple edit calls for related preferences might not guarantee atomic changes across all those changes.

Reply:
No, this class does not offer that functionality.